File: app/src/main/java/com/alexkn/syntact/app/Property.java
package com.alexkn.syntact.app;

import android.app.Application;
import android.content.Context;
import android.content.res.Resources;
import android.util.Log;

import com.alexkn.syntact.R;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class Property {

    private static final String TAG = Property.class.getSimpleName();

    private Properties properties;

    @Inject
    public Property(Context application) {

        try {
            Resources resources = application.getResources();
            InputStream inputStream = resources.openRawResource(R.raw.config);

            properties = new Properties();
            properties.load(inputStream);
        } catch (IOException e) {
            Log.e(TAG, "loadProperties: Unable to open config file.", e);
        }
    }

    public String get(String key) {

        return properties.getProperty(key);
    }
}

File: app/src/main/java/com/alexkn/syntact/app/Property.kt
package com.alexkn.syntact.app

import android.content.Context
import android.util.Log
import java.io.IOException
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class Property @Inject constructor(context: Context) {

    private var properties = Properties()

    init {
        try {
            context.assets.open("config.properties").use {
                properties.load(it)
            }
        } catch (e: IOException) {
            Log.e(TAG, "loadProperties: Unable to open config file.", e)
        }

    }

    operator fun get(key: String): String = properties.getProperty(key)
}
